Unused function compiler warning

When trying to analyze this project with [LGTM](https://lgtm.com), `make` fails due the following C++ warning that was promoted to error:

```
[2018-07-06 14:47:19] [build] g++ -Wall -Werror -O1   -c -o rpacket_entry_0103.o rpacket_entry_0103.cpp
[2018-07-06 14:47:20] [build] rpacket_entry_0103.cpp:32:10: warning: ‘size_t {anonymous}::copyMember(std::vector<unsigned char>&, size_t, const rclient::RPacketEntry_0103&)’ defined but not used [-Wunused-function]
[2018-07-06 14:47:20] [build]    size_t copyMember(RVECTORTYPE<unsigned char> &buf, const size_t i, const rclient::RPacketEntry_0103 &entry){
[2018-07-06 14:47:20] [build]           ^~~~~~~~~~
```

The full LGTM build log is included below.  The log does not indicate which version of g++ is used.  I was able to reproduce the warning on g++ 5.3.1.  The warning did not appear on g++ 4.4.4.

```
[2018-07-06 14:46:40] [build] Configuration file:
[2018-07-06 14:46:40] [build] queries:
[2018-07-06 14:46:40] [build] - exclude: "py/missing-docstring"
[2018-07-06 14:46:40] [build] - exclude: "js/similar-function"
[2018-07-06 14:46:40] [build] - exclude: "js/duplicate-function"
[2018-07-06 14:46:40] [build] extraction:
[2018-07-06 14:46:40] [build]   java:
[2018-07-06 14:46:40] [build]     index:
[2018-07-06 14:46:40] [build]       xml_mode: "ALL"
[2018-07-06 14:46:41] [build] prepare:
[2018-07-06 14:46:41] [build] configure:
[2018-07-06 14:46:41] [build] Temp dir: /tmp/deptrace.PzZoC
[2018-07-06 14:46:41] [build] Server pid: 462
[2018-07-06 14:46:41] [build] + /opt/buildtools/cpp/detect_source_root
[2018-07-06 14:46:41] [build] detect_source_root: Using build system found in '.'
[2018-07-06 14:46:42] [build] deptrace-server: Loaded database with 998987 files
[2018-07-06 14:46:42] [build] + '[' -L _lgtm_detected_source_root ']'
[2018-07-06 14:46:42] [build] ++ readlink _lgtm_detected_source_root
[2018-07-06 14:46:42] [build] + cd .
[2018-07-06 14:46:42] [build] + configure_prefix_opt=
[2018-07-06 14:46:42] [build] + '[' -f configure.ac ']'
[2018-07-06 14:46:42] [build] + '[' -f configure.in ']'
[2018-07-06 14:46:42] [build] + '[' -f CMakeLists.txt ']'
[2018-07-06 14:46:42] [build] + try_configure
[2018-07-06 14:46:42] [build] + build_dir=.
[2018-07-06 14:46:42] [build] + root_dir=.
[2018-07-06 14:46:42] [build] + grep -q 'AC_MSG_.* configure in a separate.* directory' configure.in configure.ac
[2018-07-06 14:46:42] [build] + for configure in configure configure.gnu
[2018-07-06 14:46:42] [build] + '[' -x configure ']'
[2018-07-06 14:46:42] [build] + for configure in configure configure.gnu
[2018-07-06 14:46:42] [build] + '[' -x configure.gnu ']'
[2018-07-06 14:46:42] [build] + rm -rf _lgtm_build_dir
[2018-07-06 14:46:42] [build] + '[' -f bootstrap.sh ']'
[2018-07-06 14:46:42] [build] + '[' -f bootstrap ']'
[2018-07-06 14:46:42] [build] + '[' -f autogen.sh ']'
[2018-07-06 14:46:42] [build] + try_configure
[2018-07-06 14:46:42] [build] + build_dir=.
[2018-07-06 14:46:42] [build] + root_dir=.
[2018-07-06 14:46:42] [build] + grep -q 'AC_MSG_.* configure in a separate.* directory' configure.in configure.ac
[2018-07-06 14:46:42] [build] + for configure in configure configure.gnu
[2018-07-06 14:46:42] [build] + '[' -x configure ']'
[2018-07-06 14:46:42] [build] + for configure in configure configure.gnu
[2018-07-06 14:46:42] [build] + '[' -x configure.gnu ']'
[2018-07-06 14:46:42] [build] + rm -rf _lgtm_build_dir
[2018-07-06 14:46:42] [build] + '[' -f wscript ']'
[2018-07-06 14:46:42] [build] + '[' -f Kbuild ']'
[2018-07-06 14:46:42] [build] + '[' -f configure.ac ']'
[2018-07-06 14:46:42] [build] + '[' -f configure.in ']'
[2018-07-06 14:46:42] [build] + try_configure
[2018-07-06 14:46:42] [build] + build_dir=.
[2018-07-06 14:46:42] [build] + root_dir=.
[2018-07-06 14:46:42] [build] + grep -q 'AC_MSG_.* configure in a separate.* directory' configure.in configure.ac
[2018-07-06 14:46:42] [build] + for configure in configure configure.gnu
[2018-07-06 14:46:42] [build] + '[' -x configure ']'
[2018-07-06 14:46:42] [build] + for configure in configure configure.gnu
[2018-07-06 14:46:42] [build] + '[' -x configure.gnu ']'
[2018-07-06 14:46:42] [build] + rm -rf _lgtm_build_dir
[2018-07-06 14:46:42] [build] + exit 0
[2018-07-06 14:46:42] [build] index:
[2018-07-06 14:46:42] [build] Temp dir: /tmp/deptrace.F6Y3v
[2018-07-06 14:46:42] [build] Server pid: 485
[2018-07-06 14:46:43] [build] Running '/opt/dist/tools/preload_tracer /opt/buildtools/cpp/do-build' with tracer /opt/dist/tools/preload_tracer
[2018-07-06 14:46:43] [build] + '[' -L _lgtm_detected_source_root ']'
[2018-07-06 14:46:43] [build] ++ readlink _lgtm_detected_source_root
[2018-07-06 14:46:43] [build] deptrace-server: Loaded database with 998987 files
[2018-07-06 14:46:43] [build] + cd .
[2018-07-06 14:46:43] [build] + '[' -d _lgtm_build_dir ']'
[2018-07-06 14:46:43] [build] + '[' -f SConstruct ']'
[2018-07-06 14:46:43] [build] + '[' -f wscript ']'
[2018-07-06 14:46:43] [build] + '[' -f Makefile ']'
[2018-07-06 14:46:43] [build] + make
[2018-07-06 14:46:43] [build] g++ -Wall -Werror -O1   -c -o endian_converter.o endian_converter.cpp
[2018-07-06 14:46:45] [build] deptrace-server: File '/usr/x86_64-linux-gnu/include/stdc-predef.h' found in package libc6-dev-amd64-cross
[2018-07-06 14:46:45] [build] deptrace-server: Running installation program: /opt/work/autobuild/sudo-install
[2018-07-06 14:46:45] [build] Reading package lists...
[2018-07-06 14:46:46] [build] Building dependency tree...
[2018-07-06 14:46:46] [build] Reading state information...
[2018-07-06 14:46:46] [build] The following additional packages will be installed:
[2018-07-06 14:46:46] [build]   libc6-amd64-cross linux-libc-dev-amd64-cross
[2018-07-06 14:46:46] [build] The following NEW packages will be installed:
[2018-07-06 14:46:46] [build]   libc6-amd64-cross libc6-dev-amd64-cross linux-libc-dev-amd64-cross
[2018-07-06 14:46:46] [build] 0 upgraded, 3 newly installed, 0 to remove and 0 not upgraded.
[2018-07-06 14:46:46] [build] Need to get 5,074 kB of archives.
[2018-07-06 14:46:46] [build] After this operation, 4,592 kB of additional disk space will be used.
[2018-07-06 14:46:46] [build] Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libc6-amd64-cross all 2.27-3ubuntu1cross1 [1,611 kB]
[2018-07-06 14:46:46] [build] Get:2 http://archive.ubuntu.com/ubuntu bionic/universe amd64 linux-libc-dev-amd64-cross all 4.15.0-18.19cross1 [882 kB]
[2018-07-06 14:46:46] [build] Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libc6-dev-amd64-cross all 2.27-3ubuntu1cross1 [2,582 kB]
[2018-07-06 14:46:46] [build] debconf: delaying package configuration, since apt-utils is not installed
[2018-07-06 14:46:47] [build] Fetched 5,074 kB in 0s (11.9 MB/s)
[2018-07-06 14:46:47] [build] Selecting previously unselected package libc6-amd64-cross.
[2018-07-06 14:46:47] [build] (Reading database ... 64976 files and directories currently installed.)
[2018-07-06 14:46:47] [build] Preparing to unpack .../libc6-amd64-cross_2.27-3ubuntu1cross1_all.deb ...
[2018-07-06 14:46:47] [build] Unpacking libc6-amd64-cross (2.27-3ubuntu1cross1) ...
[2018-07-06 14:46:47] [build] Selecting previously unselected package linux-libc-dev-amd64-cross.
[2018-07-06 14:46:47] [build] Preparing to unpack .../linux-libc-dev-amd64-cross_4.15.0-18.19cross1_all.deb ...
[2018-07-06 14:46:47] [build] Unpacking linux-libc-dev-amd64-cross (4.15.0-18.19cross1) ...
[2018-07-06 14:46:47] [build] Selecting previously unselected package libc6-dev-amd64-cross.
[2018-07-06 14:46:47] [build] Preparing to unpack .../libc6-dev-amd64-cross_2.27-3ubuntu1cross1_all.deb ...
[2018-07-06 14:46:47] [build] Unpacking libc6-dev-amd64-cross (2.27-3ubuntu1cross1) ...
[2018-07-06 14:46:48] [build] Setting up linux-libc-dev-amd64-cross (4.15.0-18.19cross1) ...
[2018-07-06 14:46:48] [build] Setting up libc6-amd64-cross (2.27-3ubuntu1cross1) ...
[2018-07-06 14:46:48] [build] Setting up libc6-dev-amd64-cross (2.27-3ubuntu1cross1) ...
[2018-07-06 14:46:48] [build] deptrace-server: Package installer exited with code 0
[2018-07-06 14:46:57] [build] g++ -Wall -Werror -O1   -c -o network_error.o network_error.cpp
[2018-07-06 14:46:59] [build] g++ -Wall -Werror -O1   -c -o network_manager.o network_manager.cpp
[2018-07-06 14:47:02] [build] g++ -Wall -Werror -O1   -c -o qap1_header.o qap1_header.cpp
[2018-07-06 14:47:03] [build] g++ -Wall -Werror -O1   -c -o rclient.o rclient.cpp
[2018-07-06 14:47:05] [build] g++ -Wall -Werror -O1   -c -o rexp.o rexp.cpp
[2018-07-06 14:47:07] [build] g++ -Wall -Werror -O1   -c -o rexp_double.o rexp_double.cpp
[2018-07-06 14:47:08] [build] g++ -Wall -Werror -O1   -c -o rexp_integer.o rexp_integer.cpp
[2018-07-06 14:47:10] [build] g++ -Wall -Werror -O1   -c -o rexp_list.o rexp_list.cpp
[2018-07-06 14:47:11] [build] g++ -Wall -Werror -O1   -c -o rexp_null.o rexp_null.cpp
[2018-07-06 14:47:13] [build] g++ -Wall -Werror -O1   -c -o rexp_pairlist.o rexp_pairlist.cpp
[2018-07-06 14:47:14] [build] g++ -Wall -Werror -O1   -c -o rexp_string.o rexp_string.cpp
[2018-07-06 14:47:16] [build] g++ -Wall -Werror -O1   -c -o rexp_vector.o rexp_vector.cpp
[2018-07-06 14:47:17] [build] g++ -Wall -Werror -O1   -c -o rpacket.o rpacket.cpp
[2018-07-06 14:47:19] [build] g++ -Wall -Werror -O1   -c -o rpacket_entry_0103.o rpacket_entry_0103.cpp
[2018-07-06 14:47:20] [build] rpacket_entry_0103.cpp:32:10: warning: ‘size_t {anonymous}::copyMember(std::vector<unsigned char>&, size_t, const rclient::RPacketEntry_0103&)’ defined but not used [-Wunused-function]
[2018-07-06 14:47:20] [build]    size_t copyMember(RVECTORTYPE<unsigned char> &buf, const size_t i, const rclient::RPacketEntry_0103 &entry){
[2018-07-06 14:47:20] [build]           ^~~~~~~~~~
[2018-07-06 14:47:21] [build] g++ -Wall -Werror -O1   -c -o demo.o demo.cpp
[2018-07-06 14:47:25] [build] g++ -lcrypt demo.o endian_converter.o network_error.o network_manager.o qap1_header.o rclient.o rexp.o rexp_double.o rexp_integer.o rexp_list.o rexp_null.o rexp_pairlist.o rexp_string.o rexp_vector.o rpacket.o rpacket_entry_0103.o -o demo
[2018-07-06 14:47:25] [build] rclient.o: In function `rclient::RClient::login(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
[2018-07-06 14:47:25] [build] rclient.cpp:(.text+0xc38): undefined reference to `crypt'
[2018-07-06 14:47:25] [build] collect2: error: ld returned 1 exit status
[2018-07-06 14:47:25] [build] make: *** [demo] Error 1
[2018-07-06 14:47:25] [build] Makefile:50: recipe for target 'demo' failed
[2018-07-06 14:47:25] [build] + '[' -d ../_lgtm_build_dir ']'
[2018-07-06 14:47:25] [build] Semmle autobuild: no supported build system detected.
[2018-07-06 14:47:25] [build] + for f in build build.sh
[2018-07-06 14:47:25] [build] + '[' -x build ']'
[2018-07-06 14:47:25] [build] + for f in build build.sh
[2018-07-06 14:47:25] [build] + '[' -x build.sh ']'
[2018-07-06 14:47:25] [build] + '[' -f setup.py ']'
[2018-07-06 14:47:25] [build] + echo 'Semmle autobuild: no supported build system detected.'
[2018-07-06 14:47:25] [build] + exit 1
```

--- a/rpacket_entry_0103.cpp
+++ b/rpacket_entry_0103.cpp
@@ -23,19 +23,6 @@
 
 namespace{
   
-  /** Copies data from the provided entry to the given buffer at position i
-   * @param[out] buf vector of unsigned char to copy entry data to
-   * @param[in] i position in buf to copy data
-   * @param[in] entry RPacketEntry_0103 to copy data from 
-   * @return number of bytes copied
-   */
-  size_t copyMember(RVECTORTYPE<unsigned char> &buf, const size_t i, const rclient::RPacketEntry_0103 &entry){
-    RVECTORTYPE<unsigned char> entry_ch = entry.getEntry();
-    size_t entry_length = entry.getLength();
-    memcpy(&buf[i], &entry_ch[0], entry_length);
-    return entry_length;
-  }
-
   /** Serializes integer into little-endian byte order
    * @param[out] buf To be filled with the little-endian serialized version of val
    * @param[in] val Integer to be serialized

